File: haskell/src/Shapes/Base/Commons/Displayable.hs
module Shapes.Base.Commons.Displayable (Displayable (..)) where

class Displayable a where
    shouldDisplay :: a -> Bool
    shouldDisplay _ = True

    display :: a -> IO ()
    display shape =
        if shouldDisplay shape
            then putStrLn "Yes, this shape can be displayed."
            else putStrLn "No, this shape cannot be displayed."

File: kotlin/src/main/kotlin/shapes/base/commons/Displayable.kt
package shapes.base.commons

interface Displayable {
    fun shouldDisplay(): Boolean {
        return true
    }

    fun display() {
        if (shouldDisplay()) {
            println("Yes, this shape can be displayed.")
        } else {
            println("No, this shape cannot be displayed.")
        }
    }
}
